pkg/ottl/functions.go

```diff
@@ -86,7 +160,17 @@ func (p *Parser[K]) buildArgs(ed editor, argsVal reflect.Value) error {
 		if err != nil {
 			return fmt.Errorf("invalid argument at position %v: %w", i, err)
 		}
-		field.Set(reflect.ValueOf(val))
+		if isOptional {
+			fieldVal.Set(reflect.ValueOf(val))
```

Comment:
>what happens when an optional param is not passed in, would this be nil?

Same question here.

Reply:
`presenceVal` is maybe a poor name, something like `precenseField` would be more accurate. This is the field on the `Optional` struct that indicates whether a value has been set. By default it is `false`, indicating that the user has not passed in a value for this optional parameter. If the user does pass something in, we set it to `true`.